src/arrayMethodJoin.js

```diff
@@ -5,8 +5,22 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+    const resultSeparator = separator !== undefined ? `${separator}` : ',';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        (i === this.length - 1)
+          ? result += ''
+          : result += '' + resultSeparator;
+      } else {
+        (i === this.length - 1)
+          ? result += this[i]
+          : result += this[i] + resultSeparator;
+      }
+    };
```

Comment:
I'm 100% sure you can think of an idea to simplify it! In case of any problems tag me here.